src/rules/no-cycle.js

```diff
@@ -27,6 +22,7 @@ module.exports = {
       maxDepth: {
         anyOf: [
           {
+            // This is a nop, but we can keep it for backwards-compatibility
```

Comment:
it needs to not be a no-op - if there's a flaw at a depth greater than this value, we can't report it.

Reply:
Define "flaw"?

Reply:
like if there's a cycle beyond that depth, we currently don't report it, and we need to continue not reporting it

Reply:
If we insist on backwards-compatibility, yes.

But why would users want to ignore dependency cycles that are very big? I think this was originally supposed to be a performance workaround to prevent `no-cycle` from taking a very long time in large code-bases?

Reply:
We do; breaking changes are harmful.

I agree it exists as a perf workaround and that ideally nobody should need to use it after this PR ships, which is great! but it could also be used to silence cycle warnings in dependencies, or unrelated parts of the codebase.

Reply:
👍🏻